engine/resource/src/resource.cpp

```diff
@@ -1038,28 +1042,108 @@ static const char* GetExtFromPath(const char* name, char* buffer, uint32_t buffe
 }
 
 // Assumes m_LoadMutex is already held
-static Result DoGet(HFactory factory, const char* name, void** resource)
-{
-    assert(name);
-    assert(resource);
-
-    DM_PROFILE(__FUNCTION__);
+static Result DoCreateResource(HFactory factory, SResourceType* resource_type, const char* name, const char* canonical_path,
+    dmhash_t canonical_path_hash, void* buffer, uint32_t buffer_size, void** resource_out)
+{
+    // TODO: We should *NOT* allocate SResource dynamically...
```

Comment:
Waqt struct is `SResource` ? 🤔

Reply:
I have no idea, it's an old comment. I just moved it along with the rest of the code so I don't really know what it is referring to 🤔